packages/python/plotly/plotly/basedatatypes.py

```diff
@@ -1090,42 +1145,51 @@ def _add_annotation_like(
                 "Received col parameter but not row.\n"
                 "row and col must be specified together"
             )
+        grid_ref = self._validate_get_grid_ref()
 
         # Get grid_ref if specific row or column requested
         if row is not None:
-            grid_ref = self._validate_get_grid_ref()
-            refs = grid_ref[row - 1][col - 1]
+            # TODO It is assumed that grid_ref has an equal number of columns in
```

Comment:
Yes, this can be not-true, you can have really odd layouts: https://plotly.com/python/subplots/#multiple-custom-sized-subplots

Reply:
Hey so I checked this out and it turns out even in this case `grid_ref` is still a rectangle, just some of the entries are None. So we could still easily iterate through the `grid_ref` using something like
```python
for r in range(1,len(grid_ref)+1):
    for c in range(1,len(grid_ref[0])+1):
        sub_plot=fig.get_subplot(row=r,col=c)
        if sub_plot is not None:
            ....
```
We just check if it is not None before doing anything to it :wrench: